test-suite/shared/src/test/require-jdk17/org/scalajs/testsuite/javalib/util/random/RandomGeneratorTest.scala

```diff
@@ -0,0 +1,433 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.javalib.util.random
+
+import org.junit.Assert._
+import org.junit.Test
+
+import org.scalajs.testsuite.utils.AssertThrows.assertThrows
+import org.scalajs.testsuite.utils.Platform._
+
+import java.util.random.RandomGenerator
+
+class RandomGeneratorTest {
+  /** Returns a RandomGenerator whose calls to `nextLong()` will return
+   *  the elements of `nextLongs`.
+   *
+   *  It does not loop. Requesting more elements will throw an AssertionError.
+   */
+  private def make(nextLongs: Long*): RandomGenerator = {
+    val iter = nextLongs.iterator
+    new RandomGenerator {
+      def nextLong(): Long = {
+        if (!iter.hasNext)
+          throw new AssertionError("Generator has run out")
+        iter.next()
+      }
+    }
+  }
+
+  /* 20 random Long values that we use as `nextLong` values in our tests
+   * 0xcdff356ba8f073c7L
+   * 0x4dd3ce2b9e4f98d3L
+   * 0xb85927b9d6e3249bL
+   * 0x0bf3df8c2f645e7dL
+   * 0x857f63be5f2f9ca0L
+   * 0xa12351957679f21fL
+   * 0xe01389542264f955L
+   * 0x1348eac1e0b1d30dL
+   * 0x496620384f075af6L
+   * 0xc7a71275b30dde50L
+   * 0xff5c1abdc835a88dL
+   * 0x7cd709fed6fc0e33L
+   * 0x457edabdccfc99b3L
+   * 0x867f0bae045cc0b5L
+   * 0xd9110b1c6a647a80L
+   * 0x27f8c35e876aa500L
+   * 0x700455bf2802ddacL
+   * 0x06b54104f821ddfeL
+   * 0xbf3b87070bbc6313L
+   * 0x74fa600cd5741c70L
```

Comment:
Have you considered another helper that calls make with these?

It looks like it would simplify (and strengthen) plenty of test cases below.

For example, `testNextBoolean` could become:

```scala
  val r = makeStd()
  assertEquals(true, r.nextBoolean())
  assertEquals(false, r.nextBoolean())
  // etc.
```

Reply:
It might simplify a number of tests.

However IMO it would *weaken* them. The way it's done now makes it absolutely certain that every output depends only on the provided values for `nextLong()`, and not on any "history" of what `nextLong()` produced in the past. History might be relevant to `nextGaussian()`, for example; the implementations of that one I've seen so far produce 2 values at a time, and save one for later.

The way it's written now is also a *lot* easier to debug when something fails. Since you have the direct relation between a `nextLong()` value and the `nextX()` call you're debugging. It's more difficult to relate those two if we implicitly rely on the stream of outputs.